clientconn_test.go

```diff
@@ -285,7 +285,8 @@ func TestDialWaitsForServerSettingsAndFails(t *testing.T) {
 				defer conn.Close()
 			}
 		}()
-		getMinConnectTimeout = func() time.Duration { return time.Second / 4 }
+		cleanup := setMinConnectTimeout(time.Second / 4)
```

Comment:
`minConnectTimeout` probably should have been a field in `ClientConn` with a non-exported `DialOption` to set it.  That would avoid these global-state-change things which also caused races.  I'll try to do this as a cleanup change later.  (This is just a note to myself.)

Reply:
Ack